### DISCUSSION TITLE
Using pyscript and THREE js for loading GLTF

### DISCUSSION THREAD

**Discussion by Homagn** [pyodide/pyodide#3120] [Q&A] [answered]
Hello, I was trying to load a GLTF file and animate it using THREE js but through pyscript that uses pyodide
So far I have been successful in loading a GLTF object with the materials, but I am stuck when trying to use the animation mixer 
I probably need to define mixer as a global variable, but it becomes inaccessible when using create_proxy function.

The attached code will successfully load GLTF file but ideally it should be also animating it, can someone please help

(html file)

[html.txt](https://github.com/pyodide/pyodide/files/9592030/html.txt)

(python file)
[python.txt](https://github.com/pyodide/pyodide/files/9592032/python.txt)

**Comment by ryanking13** [comment]
Could you give a fully reproducible code (Maybe a zip file or a github repo)?. For now I get this error message so I can't reproduce:

![image](https://user-images.githubusercontent.com/24893111/190881290-7c523250-973e-4a2c-aa7a-fb56abd01823.png)

**Comment by Homagn** [comment]
Hello, please find the reproducible code here
https://jsfiddle.net/3k02Lszt/
(please go through the portion under py-script tag as that is the main thing)
The thing Im trying to do is summarized in line 215 of the HTML file as a comment

**Comment by ryanking13** [comment, marked as answer]
Here is what I did: https://jsfiddle.net/tcofgu0a/1/ (To be honest, I know nothing about Three.js so I am not sure this is what you want)

Note that since `loader.load` is asynchronous, you will have to wait until it finishes and call `gltf_func`

So I did something like this do make sure animations are loaded from the gltf file:

```py
_animations = None
_done = False
def gltf_func(gltf): #goes inside this function only once
  #print("In gltf function ")
  global _animations
  global _done
  fileAnimations = gltf.animations
  

  #scale the whole model
  gltf.scene.scale.set(0.5, 0.5, 0.5)
  #translate the whole model downwards
  gltf.scene.position.y = -0.5
  gltf.scene.traverse(create_proxy(child_func))

  #scene.add(gltf.scene)
  loaded_model.add(gltf.scene)
  _animations = gltf.animations
  _done = True
```

```py
import asyncio
while not _done:
  await asyncio.sleep(1)
clip = THREE.AnimationClip.findByName(_animations, 'idle')
idle = mixer.clipAction(clip);
idle.play();
```

**Reply by Homagn** [reply]
This is awesome ! 
This is exactly what I wanted, Thank you so much !
There is actually no reference/videos on how to make 3D animations using pyscript and THREE js, its very new apparently
Because of your help now I can make 3D games for browser entirely using python instead of javascript :)
